### PR TITLE
refactor(AppDelegate): Migrate application:openUrl:options

### PR DESCRIPTION
**Pull request by bchrisarriola**
Also created `BitcoinURLPayload` + unit tests to encapsulate the URL request "bitcoin://". I still need to migrate `showType` which I'll do in another PR.

### CODE REVIEW

File: BlockchainTests/URL/BitcoinURLPayloadTests.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    func testBitcoinAddressInQueryArg() {
+        let address = "bitcoinaddress"
+        let url = URL(string: "bitcoin://?address=\(address)")
+        let payload = BitcoinURLPayload(url: url!)
+        XCTAssertNotNil(payload)
+        XCTAssertEqual(address, payload!.address)
+    }
+
```
Comment (woudini) [inline BlockchainTests/URL/BitcoinURLPayloadTests.swift]: Should have a test with address and amount, e.g. `175tWpb8K1S7NmH4Zx6rewF9WQrcZv245W?amount=20.3`

File: Blockchain/RootService.h [hunk 2]
```diff
@@ -221,9 +221,9 @@
 - (void)scanPrivateKeyForWatchOnlyAddress:(NSString *)address;
 - (void)askUserToAddWatchOnlyAddress:(NSString *)address success:(void (^)(NSString *))success;
 
-- (void)verifyTwoFactorSMS;
-- (void)verifyTwoFactorGoogle;
-- (void)verifyTwoFactorYubiKey;
+//- (void)verifyTwoFactorSMS;
+//- (void)verifyTwoFactorGoogle;
+//- (void)verifyTwoFactorYubiKey;
```
Comment (woudini) [inline Blockchain/RootService.h]: Not sure why two factor changes are in this PR

Comment (bchrisarriola) [reply Blockchain/RootService.h]: Actually meant to comment these out when I migrated those methods out of `RootService`—these are no longer used and I've been keeping track off what remains to be migrated by commenting stuff out of RootService.h/.m

File: BlockchainTests/URL/BitcoinURLPayloadTests.swift (hunk 1 above)
Comment (bchrisarriola) [reply BlockchainTests/URL/BitcoinURLPayloadTests.swift]: Will add a test for that

**Comment by bchrisarriola** [comment]
@woudini it was meant to be me merged into `swift`. I choose a different base to avoid the issue we saw the other day about potential duplicate methods but should've communicated that in the PR description.

I'll `cherry-pick` into `swift` and delete `app_lifecycle`.

**Pull request merged**